public/js/hits.js

```diff
@@ -0,0 +1,276 @@
+/* eslint-disable no-unused-vars */
+import { Component } from 'react';
+import _ from 'underscore';
+
+import { ReportQuery } from './query';
+import Hit from './hit';
+import HSP from './hsp';
+import AlignmentExporter from './alignment_exporter';
+/* eslint-enable no-unused-vars */
+
+class Hits extends Component {
+    constructor(props) {
+        super(props);
+        this.numUpdates = 0;
+        this.nextQuery = 0;
+        this.nextHit = 0;
+        this.nextHSP = 0;
+        this.maxHSPs = 3; // max HSPs to render in a cycle
+        this.state = props.state;
+        this.prepareAlignmentOfSelectedHits = this.prepareAlignmentOfSelectedHits.bind(this);
+    }
+
+    componentDidMount() {
+        this.componentDidUpdate(this.props, this.state);
+    }
+
+    /**
+    * Called for the first time after as BLAST results have been retrieved from
+    * the server and added to this.state by fetchResults. Only summary overview
+    * and circos would have been rendered at this point. At this stage we kick
+    * start iteratively adding 1 HSP to the page every 25 milli-seconds.
+    */
+    componentDidUpdate(prevProps, prevState) {
+        // Log to console how long the last update take?
+        // console.log((Date.now() - this.lastTimeStamp) / 1000);
+
+        // Lock sidebar in its position on the first update.
+        if (this.nextQuery == 0 && this.nextHit == 0 && this.nextHSP == 0) {
+            this.affixSidebar();
+        }
```

Comment:
Should this be a responsibility of report.js? I think now with the custom hits batch-rendering logic separated, we should be able to affix the sidebar one in report.js without any of the batch-rendering related checks or am I missing something?

Reply:
@tadast from my perspective it's better this way

in report js, we already have 2 checks
- if a warning present
- if results available